src/executor/evaluate/mod.rs

```diff
@@ -281,5 +283,53 @@ async fn evaluate_function<'a, T: 'static + Debug>(
 
             Ok(Evaluated::from(Value::Str(converted)))
         }
+        Function::Ceil(expr) => {
+            let number = match eval(expr).await?.try_into()? {
+                Value::F64(number) => Some(number),
+                Value::Str(s) => match f64::from_str(&s) {
+                    Ok(f) => Some(f),
+                    Err(_) => None,
+                },
+                Value::I64(number) => f64::from_i64(number),
+                _ => None,
+            };
+
+            match number {
+                Some(number) => Ok(Evaluated::from(Value::F64(number.ceil()))),
+                None => Err(EvaluateError::FunctionRequiresFloatValue("CEIL".to_owned()).into()),
+            }
+        }
+        Function::Round(expr) => {
+            let number = match eval(expr).await?.try_into()? {
+                Value::F64(number) => Some(number),
+                Value::Str(s) => match f64::from_str(&s) {
+                    Ok(f) => Some(f),
+                    Err(_) => None,
+                },
+                Value::I64(number) => f64::from_i64(number),
+                _ => None,
+            };
+
+            match number {
+                Some(number) => Ok(Evaluated::from(Value::F64(number.round()))),
+                None => Err(EvaluateError::FunctionRequiresFloatValue("ROUND".to_owned()).into()),
+            }
+        }
+        Function::Floor(expr) => {
+            let number = match eval(expr).await?.try_into()? {
+                Value::F64(number) => Some(number),
+                Value::Str(s) => match f64::from_str(&s) {
+                    Ok(f) => Some(f),
+                    Err(_) => None,
+                },
```

Comment:
Let's handle `Str` input as an error `FunctionRequiresFloatValue`.
We can accept, I64 to F64 because it is also natively supported by Rust lang itself, but Str is not.
Users who need to run `FLOOR` using `Str` value, then they can use `CAST` function.

It is also better not to convert `Err` into non `Err` types.
ref https://github.com/gluesql/gluesql/pull/289#discussion_r690053201